src/app/tests/MockAclTestContext.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/tests/MockAclTestContext.h>
+
+#include <access/AccessControl.h>
+#include <access/examples/PermissiveAccessControlDelegate.h>
+#include <app/InteractionModelEngine.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/ErrorStr.h>
+
+namespace {
+using namespace chip;
+using namespace chip::Access;
+using Entry                     = chip::Access::AccessControl::Entry;
+using EntryIterator             = chip::Access::AccessControl::EntryIterator;
+using Target                    = Entry::Target;
+chip::ClusterId kTestClusterId1 = 6;
+
+class TestDeviceTypeResolver : public AccessControl::DeviceTypeResolver
+{
+public:
+    bool IsDeviceTypeOnEndpoint(DeviceTypeId deviceType, EndpointId endpoint) override { return false; }
+} gDeviceTypeResolver;
+
+AccessControl gTestAccessControl;
+
+class TestAccessControlDelegate : public AccessControl::Delegate
+{
+public:
+    CHIP_ERROR Init() override { return CHIP_NO_ERROR; }
+    void Finish() override {}
+
+    // Capabilities
+    CHIP_ERROR GetMaxEntryCount(size_t & value) const override
+    {
+        value = 0;
+        return CHIP_NO_ERROR;
+    }
+
+    // Actualities
+    CHIP_ERROR GetEntryCount(size_t & value) const override
```

Comment:
Why provide an implementation for this that isn't different from the base class's impl? Isn't just `Check` sufficient to implement here?

Reply:
yes, check is enough. removed this file.